src/main/java/mekanism/client/render/RenderTickHandler.java

```diff
@@ -353,44 +359,85 @@ public void onBlockHover(DrawSelectionEvent.HighlightBlock event) {
             profiler.pop();
 
             ItemStack stack = player.getMainHandItem();
-            if (stack.isEmpty() || !(stack.getItem() instanceof ItemConfigurator)) {
-                //If we are not holding a configurator, look if we are in the offhand
-                stack = player.getOffhandItem();
+
+            if (rayTraceResult.getType() == Type.BLOCK && stack.getItem() instanceof ItemMekaTool tool && tool.isModuleEnabled(stack, MekanismModules.BLASTING_UNIT) && !player.isShiftKeyDown()) {
+                drawAdditionalBlockOutlines(event, player, ModuleBlastingUnit.findPositions(world, rayTraceResult.getBlockPos(), rayTraceResult.getDirection(), tool.getModule(stack, MekanismModules.BLASTING_UNIT).getCustomInstance().getBlastRadius()));
+            } else {
```

Comment:
I feel like instead of making this an `if else` this should just be an if before (or maybe after?)  and just skip the target block and instead of the implementation that is done below:
- add an AT for `LevelRenderer#renderShape`
- Do a brief call chain like in `LevelRenderer` where `ForgeHooksClient#onDrawHighlight` is done for blocks so that if any mods wants to override how the selection box or thing display for their block they can. (For example some of Mekanism's ones with the `AttributeCustomSelectionBox` have their overrides because the base shape rendering code destroys performance when trying to render them.

Reply:
The reason this isn't an if lower down is that the existing logic is cancelling the event and I didn't really want to mess with anything it was trying to do. And honestly I didn't take the time to fully understand everything this method is doing, I can take another look though.

The reason I just copied out the logic was I knew you didn't like adding mixins and it was a small method (and immersive is doing it with a mixin), but an AT should work fine.

Isn't onDrawHighlight posting the event this is working off of now? Or were you saying post a custom event?

Reply:
Yes this is the event it is working off of now so I guess some boolean would have to be set to say it is being ran (so that it doesn't loop and keep for each one drawing the AOE). But that way if say the AOE then intersects say a gas burning generator it will properly have things be at a diagonal, or when intersecting with a digital miner the FPS won't get tanked because of how vanilla's wireframe renders.

Reply:
So I figured out the gist of this method and implemented it closer to how you were thinking. I didn't bother profiling (or even checking fps) before or after so I have no idea if it improved performance noticeably but it does look much nicer.